src/util/windows.jl

```diff
@@ -0,0 +1,66 @@
+"""
+An iterator which moves across a container, as a sliding window.
+"""
+immutable EachWindowIterator{T <: AbstractArray}
```

Comment:
Yes, you can use `T<:Union{Sequence,AbstractArray,AbstractString}` as a type constraint.